frontend/src/scenes/notebooks/Notebook/Editor.tsx

```diff
@@ -182,6 +184,49 @@ export function Editor({
 
             onCreate({
                 getJSON: () => editor.getJSON(),
+                getText: () => {
+                    const titleSerializer: TextSerializer = (props): string => props.node.attrs?.title || ''
+                    const customNodeTextSerializers: Record<NotebookNodeType, TextSerializer> = {
+                        'ph-backlink': titleSerializer,
+                        'ph-early-access-feature': titleSerializer,
+                        'ph-experiment': titleSerializer,
+                        'ph-feature-flag': titleSerializer,
+                        'ph-feature-flag-code-example': titleSerializer,
+                        'ph-image': (props: { node: ProseMirrorNode }): string => {
+                            return props.node.attrs?.file?.name || ''
```

Comment:
these text serializers should probably be in another file and referenced here...

and should be typed more nicely but its a first step